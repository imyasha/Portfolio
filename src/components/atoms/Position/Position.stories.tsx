import { Position, PositionProps } from './Position';

export default {
	title: 'atoms/Position',
	component: Position,
};

export const PositionStory = (args: PositionProps) => <Position {...args} />;

PositionStory.storyName = 'Position';
PositionStory.args = {
	job: {
		company: 'Holland & Barrett',
		jobTitle: 'Software Development Engineer 2',
		startDate: '2020-08-10',
		endDate: null,
		place: 'Nuneaton, England, United Kingdom',
		description: `* Joined Holland & Barrett as a senior software engineer, first with front-end expertise in the fulfilment squad (alongside six back-end engineers)
* Led hiring process of several further UI engineers across multiple squads
* Tech lead developing a custom-built customer services portal based on React and Next.js using TypeScript and Redux, retiring a third-party solution, saving a 6-digit sum in licence fees
* Increased non-existent test-coverage to 90+%
* Introduced e2e tests using Cypress`,
		technologies: [
			'TypeScript',
			'React',
			'Next.js',
			'Redux',
			'GraphQL',
			'REST API',
			'Jest',
			'Cypress',
		],
	},
};
