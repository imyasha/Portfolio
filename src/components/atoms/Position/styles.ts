import styled from 'styled-components';

const StyledPosition = styled.div``;

const Flex = styled.div`
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	margin-bottom: 0.5rem;

	@media screen and (min-width: 768px) {
		flex-direction: row;
	}
`;

const Text = styled.div`
	margin: 0.5rem 0 3rem;

	ul {
		margin: 0;
		list-style: square;
		padding-left: 1rem;

		li {
			margin-bottom: 0.5rem;
		}
	}
`;

export { Flex, StyledPosition, Text };
