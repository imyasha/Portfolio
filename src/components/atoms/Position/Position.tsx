import { FC } from 'react';
import { format } from 'date-fns';
import { IJob } from '@Types';
import Markdown from 'markdown-to-jsx';
import { Flex, StyledPosition, Text } from './styles';

export interface PositionProps {
	job: IJob;
}

const Position: FC<PositionProps> = ({ job }) => {
	return (
		<StyledPosition>
			<Flex>
				<div>
					<strong>{job.jobTitle}</strong> -{' '}
					{job.website ? (
						<a href={job.website} target="_blank" rel="noopener noreferrer">
							{job.company}
						</a>
					) : (
						job.company
					)}
				</div>
				<div>
					{format(new Date(job.startDate), 'MM/yyyy')} -{' '}
					{job.endDate ? format(new Date(job.endDate), 'MM/yyyy') : 'present'}
				</div>
			</Flex>
			<small>
				<em>
					<strong>Technologies:</strong> {job.technologies.join(', ')}
				</em>
			</small>
			<Text>
				<Markdown>{job.description}</Markdown>
			</Text>
		</StyledPosition>
	);
};

export { Position };
