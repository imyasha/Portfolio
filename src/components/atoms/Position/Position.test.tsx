import { Position } from '../Position';
import { render } from '@testing-library/react';

describe('Position', () => {
	it('renders correctly', () => {
		const { container } = render(
			<Position
				job={{
					company: 'Holland & Barrett',
					jobTitle: 'Software Development Engineer 2',
					startDate: '2020-08-10',
					endDate: null,
					place: 'Nuneaton, England, United Kingdom',
					description: `* Joined Holland & Barrett as a senior software engineer, first with front-end expertise in the fulfilment squad (alongside six back-end engineers)
* Led hiring process of several further UI engineers across multiple squads
* Tech lead developing a custom-built customer services portal based on React and Next.js using TypeScript and Redux, retiring a third-party solution, saving a 6-digit sum in licence fees
* Increased non-existent test-coverage to 90+%
* Introduced e2e tests using Cypress`,
					technologies: [
						'TypeScript',
						'React',
						'Next.js',
						'Redux',
						'GraphQL',
						'REST API',
						'Jest',
						'Cypress',
					],
				}}
			/>
		);
		expect(container).toMatchSnapshot();
	});
});
