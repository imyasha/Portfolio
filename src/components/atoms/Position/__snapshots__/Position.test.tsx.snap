// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Position renders correctly 1`] = `
<div>
  <div
    class="styles__StyledPosition-sc-1clk440-0"
  >
    <div
      class="styles__Flex-sc-1clk440-1 gzqcjV"
    >
      <div>
        <strong>
          Software Development Engineer 2
        </strong>
         -
         
        Holland & Barrett
      </div>
      <div>
        08/2020
         -
         
        present
      </div>
    </div>
    <small>
      <em>
        <strong>
          Technologies:
        </strong>
         
        TypeScript, React, Next.js, Redux, GraphQL, REST API, Jest, Cypress
      </em>
    </small>
    <div
      class="styles__Text-sc-1clk440-2 fLfbMf"
    >
      <ul>
        <li>
          Joined Holland & Barrett as a senior software engineer, first with front-end expertise in the fulfilment squad (alongside six back-end engineers)
        </li>
        <li>
          Led hiring process of several further UI engineers across multiple squads
        </li>
        <li>
          Tech lead developing a custom-built customer services portal based on React and Next.js using TypeScript and Redux, retiring a third-party solution, saving a 6-digit sum in licence fees
        </li>
        <li>
          Increased non-existent test-coverage to 90+%
        </li>
        <li>
          Introduced e2e tests using Cypress
        </li>
      </ul>
    </div>
  </div>
</div>
`;
